pymc3/distributions/continuous.py

```diff
@@ -3836,27 +3835,27 @@ class Interpolated(BoundedContinuous):
         Probability density function evaluated on lattice ``x_points``
     """
 
-    def __init__(self, x_points, pdf_points, *args, **kwargs):
-        self.lower = lower = at.as_tensor_variable(x_points[0])
-        self.upper = upper = at.as_tensor_variable(x_points[-1])
+    rv_op = interpolated
 
-        super().__init__(lower=lower, upper=upper, *args, **kwargs)
+    @classmethod
+    def dist(cls, x_points, pdf_points, *args, **kwargs):
+
+        lower = at.as_tensor_variable(x_points[0])
+        upper = at.as_tensor_variable(x_points[-1])
 
         interp = InterpolatedUnivariateSpline(x_points, pdf_points, k=1, ext="zeros")
         Z = interp.integral(x_points[0], x_points[-1])
 
-        self.Z = at.as_tensor_variable(Z)
+        Z = at.as_tensor_variable(Z)
         self.interp_op = SplineWrapper(interp)
```

Comment:
the `interp_op` will not be reached from the`logp`, you will have to define it in there.